ngraph/frontend/paddlepaddle/include/paddlepaddle_frontend/place.hpp

```diff
@@ -17,47 +17,171 @@
 #pragma once
 
 #include <frontend_manager/frontend_manager.hpp>
+// #include "framework.pb.h"
 
 namespace ngraph {
-namespace frontend {
-
-class PlacePDPD : public Place
-{
-    // TODO
-};
-
-class InPortPlacePDPD : public PlacePDPD
-{
-    // TODO
-};
-
-class OutPortPlacePDPD : public PlacePDPD
-{
-    // TODO
-};
-
-class VarPlacePDPD;
-class OpPlacePDPD : public PlacePDPD
-{
-public:
-    InputModel* model;
-    const void* op; // TODO: make it cleaner
-    std::map<std::string, std::vector<std::weak_ptr<VarPlacePDPD>>> outputs;
-    std::map<std::string, std::vector<std::weak_ptr<VarPlacePDPD>>> inputs;
-    OpPlacePDPD(const void* _op) : op(_op) {}
-};
-
-class VarPlacePDPD : public PlacePDPD
-{
-public:
-    InputModel* model;
-    const void* var; // TODO: make it cleaner
-    std::vector<std::weak_ptr<OpPlacePDPD>> producing_ops; // should never have more than 1 element
-    std::vector<std::weak_ptr<OpPlacePDPD>> consuming_ops;
-    PartialShape shape;
-    element::Type type;
-    VarPlacePDPD(const void* _var) : var(_var) {}
-};
-
-} // namespace frontend
+    namespace frontend {
+
+        class PDPDPlace : public Place {
+        public:
+            PDPDPlace(const InputModel& input_model, const std::vector<std::string>& names)
+                    : m_input_model(input_model),
+                      m_names(names) {
+            }
+
+            explicit PDPDPlace(const InputModel& input_model) : PDPDPlace(input_model, {}) {
+            }
+
+            ~PDPDPlace() override = default;
+
+            bool isInput() const override;
+
+            bool isOutput() const override;
+
+            bool isEqual(Ptr another) const override { return this == another.get(); }
+
+            std::vector<std::string> getNames() const override { return m_names; }
+
+        private:
+            std::vector<std::string> m_names;
+            const InputModel& m_input_model;
+        };
+
+        class TensorPlacePDPD;
+        class OpPlacePDPD;
+
+        class InPortPlacePDPD : public PDPDPlace {
+        public:
+            InPortPlacePDPD(const InputModel& input_model,
+                            const std::weak_ptr<TensorPlacePDPD>& from_tensor,
+                            const std::weak_ptr<OpPlacePDPD>& consuming_op)
+                    : PDPDPlace(input_model),
+                      m_tensor(from_tensor),
+                      m_consuming_op(consuming_op) {
+            }
+
+        private:
+            std::weak_ptr<TensorPlacePDPD> m_tensor;
```

Comment:
Since these are private members, it is needed to have getters for m_tensor and m_consuming_op.

Reply:
done